tests/queue/queue_shortcuts_usm.h

```diff
@@ -0,0 +1,291 @@
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#ifndef SYCL_CTS_QUEUE_QUEUE_SHORTCUTS_USM_H
+#define SYCL_CTS_QUEUE_QUEUE_SHORTCUTS_USM_H
+
+#include <utility>
+
+#include "../common/common.h"
+#include "../common/disabled_for_test_case.h"
+#include "../common/get_cts_object.h"
+
+namespace queue_shortcuts_usm {
+
+template <typename T>
+struct runner_memcpy {
+  runner_memcpy(sycl::queue q, unsigned int count)
+      : queue(q), element_count(count) {}
+
+  template <typename kernel_helper>
+  sycl::event run_memcpy_copy(kernel_helper helper) {
+    // set up initial values on host
+    const T t_init{0};
+    const T t_test{1};  // to initialize the sequence: t_test, ++t_test, etc.
+    std::unique_ptr<T[]> h_src = std::make_unique<T[]>(element_count);
+    std::iota(h_src.get(), h_src.get() + element_count, t_test);
+    std::unique_ptr<T[]> h_dest = std::make_unique<T[]>(element_count);
+    std::fill(h_dest.get(), h_dest.get() + element_count, t_init);
+
+    // set initial values of device-allocated usm buffer without using shortcuts
+    T* d_src = sycl::malloc_device<T>(element_count, queue);
+    T* d_dest = sycl::malloc_device<T>(element_count, queue);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(d_src, h_src.get(), element_count * sizeof(T));
+      cgh.memcpy(d_dest, h_dest.get(), element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // perform the copy from d_src to d_dest using shortcuts
+    sycl::event event = helper(d_dest, d_src);
+    event.wait();
+
+    // copy destination buffer back to host without shortcut functions
+    std::unique_ptr<T[]> h_actual = std::make_unique<T[]>(element_count);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(h_actual.get(), d_dest, element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // check the result
+    for (unsigned int i = 0; i < element_count; i++) {
+      CHECK(((t_test + i) == h_actual[i]));
+    }
+
+    sycl::free(d_dest, queue);
+    sycl::free(d_src, queue);
+
+    return event;
+  }
+
+  template <typename kernel_helper>
+  sycl::event run_memset(kernel_helper helper) {
+    constexpr char init = 0;
+    constexpr char test = 1;
+
+    // set up initial values on host
+    std::unique_ptr<T[]> h_ptr = std::make_unique<T[]>(element_count);
+    std::memset(reinterpret_cast<void*>(h_ptr.get()), init,
+                element_count * sizeof(T));
+
+    // set initial values of device-allocated usm buffer without using shortcuts
+    T* d_ptr = sycl::malloc_device<T>(element_count, queue);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(d_ptr, h_ptr.get(), element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // perform the memset from d_src to d_dest using shortcuts
```

Comment:
There are no `d_src` and `d_dest` variables in this function.

Reply:
I updated the comments.